chemprop/cli/predict.py

```diff
@@ -302,14 +152,23 @@ def process_predict_args(args: Namespace) -> Namespace:
 
 def validate_predict_args(args):
     # TODO: once args.checkpoint_dir and args.checkpoint are consolidated, need to change this as well. Not able to make this required in common.py as it may not be provided for training.
-    if args.checkpoint_path is None:
+    if args.checkpoint is None:
         raise ValueError("Must provide a checkpoint path for prediction.")
 
 
 def main(args):
-    model = MPNN.load_from_checkpoint(args.checkpoint_path)
 
-    bond_messages = isinstance(model.message_passing, BondMessagePassing)
+    if args.smiles_columns is None and args.reaction_columns is None:
+        n_components = 1
+    else:
+        n_components = len(args.smiles_columns) if args.smiles_columns is not None else 0
+        n_components += len(args.reaction_columns) if args.reaction_columns is not None else 0
```

Comment:
```suggestion
    match (args.smiles_columns, args.reaction_columns):
        case [None, None]:
            n_components = 1
        case [_, None]:
            n_components = len(args.smiles_columns)
        case [None, _]:
            n_components = len(args.reaction_columns)
        case _:
            n_components = len(args.smiles_columns) + len(args.reaction_columns)
```
IMO this is cleaner to me. At the very least, split it up into `if`/`elif`/`else`

Reply:
I love this! Thanks for the suggestion!